docs/feature_encoders.md

```diff
@@ -158,6 +158,21 @@ bool encoder_update_user(uint8_t index, bool clockwise) {
 }
 ```
 
+To read an encoder that is not directly connected to an input pin but to another interface, implement `encoder_read_state_user` (the default implmentation just returns the pin state):
+
+```c
+uint8_t encoder_read_state_user(uint8_t index) {
+    ...
+}
+```
+
+Initialization codes should be implmented in `encoder_init_state_user`. Leave it empty if unneded by `encoder_read_state_user` to avoid standard GPIO pin initialization.
+
+```c
+void encoder_init_state_user(uint8_t index) {
+}
+```
```

Comment:
````suggestion
Initialization code should be implemented in `encoder_init_state_kb`:

```c
void encoder_init_state_kb(uint8_t index) {
    // set up encoder at the required index
}

To read an encoder that is not directly connected to an input pin but to another interface, implement `encoder_read_state_kb`:

```c
uint8_t encoder_read_state_kb(uint8_t index) {
    return ... ; // This should return `-1`, `0`, or `1`, to match standard encoder "Gray Code" outputs.
}
```

```
````